ui/perfherder/alerts/AlertsView.jsx

```diff
@@ -0,0 +1,324 @@
+import {
+  Alert,
+  Container,
+  Row,
+  Pagination,
+  PaginationItem,
+  PaginationLink,
+} from 'reactstrap';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCog } from '@fortawesome/free-solid-svg-icons';
+
+import perf from '../../js/perf';
+import withValidation from '../Validation';
+import { convertParams, getFrameworkData, getStatus } from '../helpers';
+import { summaryStatusMap, endpoints } from '../constants';
+import { createQueryParams, getApiUrl } from '../../helpers/url';
+import { getData, processResponse } from '../../helpers/http';
+import ErrorMessages from '../../shared/ErrorMessages';
+import OptionCollectionModel from '../../models/optionCollection';
+import {
+  genericErrorMessage,
+  errorMessageClass,
+} from '../../helpers/constants';
+import ErrorBoundary from '../../shared/ErrorBoundary';
+
+import AlertsViewControls from './AlertsViewControls';
+
+// TODO remove $stateParams and $state after switching to react router
+export class AlertsView extends React.Component {
+  constructor(props) {
+    super(props);
+    this.validated = this.props.validated;
+    this.state = {
+      status: this.getDefaultStatus(),
+      framework: getFrameworkData(this.validated),
+      page: this.validated.page ? parseInt(this.validated.page, 10) : 1,
+      errorMessages: [],
+      alertSummaries: [],
+      issueTrackers: [],
+      loading: false,
+      optionCollectionMap: null,
+      count: 0,
+      id: this.validated.id,
+      filteredResults: [],
+      bugTemplate: null,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchAlertSummaries();
+  }
+
+  getDefaultStatus = () => {
+    const { validated } = this.props;
+    const statusParam = convertParams(validated, 'status');
+    if (!statusParam) {
+      return Object.keys(summaryStatusMap)[1];
+    }
+    return getStatus(parseInt(validated.status, 10));
+  };
+
+  updateFramework = selection => {
+    const { frameworks, updateParams } = this.props.validated;
+    const framework = frameworks.find(item => item.name === selection);
+
+    updateParams({ framework: framework.id });
+    this.setState({ framework, bugTemplate: null }, () =>
+      this.fetchAlertSummaries(),
+    );
+  };
+
+  updateStatus = status => {
+    const statusId = summaryStatusMap[status];
+    this.props.validated.updateParams({ status: statusId });
+    this.setState({ status }, () => this.fetchAlertSummaries());
+  };
+
+  navigatePage = page => {
+    const { framework, status } = this.state;
+    this.props.$state.go('alerts', {
+      page,
+      framework: framework.id,
+      status: summaryStatusMap[status],
+    });
+  };
+
+  generatePages = (page, count) => {
+    const pages = [];
+    if (page < count) {
+      for (let num = page; num < page + 5; num++) {
+        pages.push(num);
+      }
+    }
+    return pages;
+  };
+
+  // TODO potentially pass as a prop for testing purposes
```

Comment:
I agree with this spirit of this suggestion! In general I have found it better to put business logic in independently-testable functions, rather than in views. Then just have react views call out to them where needed. Makes writing and understanding jest tests much easier.

Reply:
I've found there are limitations to the user-simulated react-testing-library so when there's time to write more tests, I'm sure some bits of logic will be pulled out of views and into helpers as needed to write unit tests.